message-index/messages/GHC-69925/index.md

```diff
@@ -0,0 +1,10 @@
+---
+title: Illegal unboxed string literal in pattern
+summary: Illegal unboxed string literal in pattern
+severity: error
+introduced: 9.6.1
+---
+
+When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. `"foo"#` is of type `Addr#` -- that is, "An arbitrary machine address assumed to point outside the garbage-collected heap.".
```

Comment:
```suggestion
When pattern matching, unboxed string literals (with the `MagicHash` postfix) are not allowed. Unboxed string literals are essentially C strings allocated outside of the Haskell heap, and they evaluate to pointers to the resulting strings. Thus,`"foo"#` is of type `Addr#`. While `eqAddr#` can be used to compare these pointers, two identically-written unboxed string literals typically do not point the same address.".
```

Reply:
I think the fix in the example should be to use boxed string literals instead :-)